### PR TITLE
feat(firestore): async snippets

### PR DESCRIPTION
**Pull request by rafilong**
Add snippets for new [async Firestore interface](https://github.com/googleapis/python-firestore/tree/v2-staging). Snippets are ported from `firestore/cloud-client` to `firestore/cloud-async-client`.

### CODE REVIEW

**Comment by rafilong** [comment]
Force push to overwrite history and make review easier. 6fb670d752c7719a46d45fc6455339689e5a7761 directly copies sync snippets to async directory, and 495ae796ff395a5b517ab9ccd47d40b5f705131f converts to async. Would recommend reviewing the latter for `snippets*.py`.

**Comment by rafilong** [comment]
Question: are annotations scoped to the repo, project, or file? I assumed the latter, so snippets tags are shared between sync and async.

**Comment by crwilcox** [comment]
We don't have a pattern yet but I feel like appending async to the region tag makes sense as it would group with the other sample. Still haven't settled on how to display these though. For instance, should they both show under python or do we want a new Python (async) tab similar to how Java has the Android version.

Either way we need to settle on a new region tag. How are the Android ones separated?

**Comment by BenWhitehead** [comment]
It looks like in Android the java/kotlin separation is handled by the sample registry indirection (though I'm not familiar with how that actually works).

I'm okay with having an `_async` suffix if that's how we want to do it. Though, I'm not sure of the best way to do this. I know lots of snippets are managed lots of different ways.

**Comment by rafilong** [comment]
@BenWhitehead @crwilcox would adding an `_async` suffix as a temporary stopgap so we can merge this in reasonable? It should be a relatively quick fix in the future to implement a better solution, and a suffix seems fairly reasonable.

**Comment by BenWhitehead** [comment]
Adding the `_async` suffix SGTM

**Comment by rafilong** [comment]
@BenWhitehead @crwilcox suffix added!

File: firestore/cloud-async-client/snippets_test.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
+# Copyright 2017, Google, Inc.
```
Comment (crwilcox) [inline firestore/cloud-async-client/snippets_test.py:1]: I think there may be enough change to rev this?

File: firestore/cloud-async-client/snippets.py [hunk 2]
```diff
@@ -0,0 +1,774 @@
...
+    # [START structure_doc_ref_alternate_async]
+    a_lovelace_ref = db.document("users/alovelace")
+    # [END structure_doc_ref_alternate_async]
+
+    return a_lovelace_ref
+
+
+async def structure_subcollection_ref():
+    db = firestore.AsyncClient()
+    # [START structure_subcollection_ref_async]
```
Comment (crwilcox) [inline firestore/cloud-async-client/snippets.py:290]: this sample seems to be identical to sync? Do you have an idea of how common this is?

File: firestore/cloud-async-client/snippets_test.py (hunk 1 above)
Comment (busunkim96) [reply firestore/cloud-async-client/snippets_test.py:1]: +1, I think there are enough differences to bump this to 2020

Comment (rafilong) [reply firestore/cloud-async-client/snippets_test.py:1]: Resolved!

File: firestore/cloud-async-client/snippets.py (hunk 2 above)
Comment (rafilong) [reply firestore/cloud-async-client/snippets.py:290]: Not terribly common. A couple snippets just create documents/collections locally, of which there are no async calls. I would guess less than 10, maybe around 5. Should we remove these? I felt as though leaving them to illustrate how local creation is the same between sync / async was useful.

**Pull request merged**